npc.cpp

```diff
@@ -53,24 +53,43 @@ void Game::CreateLMons()
 
 void Game::GrowWMons(int rate)
 {
+	printf("\n\n GrowWMons 1: levels: %d \n\n", regions.numLevels);
 	//
 	// Now, go through each 8x8 block of the map, and make monsters if
 	// needed.
 	//
 	int level;
 	for (level = 0; level < regions.numLevels; level++) {
 		ARegionArray *pArr = regions.pRegionArrays[level];
+
+		printf("\n\n LEVEL: %d. X: %d. Y: %d \n\n", level, pArr->x,  pArr->y);
+
 		int xsec;
-		for (xsec=0; xsec< pArr->x / 8; xsec++) {
-			for (int ysec=0; ysec< pArr->y / 16; ysec++) {
+		int blockX;
+		int blockY;
+		if (pArr->x < 8) {
+			blockX = pArr->x;
+		} else {
+			blockX = 8;
+		}
+
+		if (pArr->y < 16) {
+			blockY = pArr->y;
+		} else {
+			blockY = 16;
+		}
+
+		for (xsec=0; xsec < pArr->x / blockX; xsec++) {
```

Comment:
Everywhere inside this loop where you're using xsec and ysec, they're multiplied by blockX and blockY.  So why not:
for (xsec = 0; xsec < pArr->x; xsec += blockX)
...and the same for ysec?

Reply:
and if you did that, you'd always go through at least once anyway, so you could probably simplify this change considerably.  If you changed the outermost for loop (lines 64-65 in the original) to:
for (xsec=0; xsec< pArr->x; xsec+=8) {
    for (int ysec=0; ysec< pArr->y; ysec+=16) {
...then you'd need to put a guard on the line that does the GetRegion to make sure the coordinates are in bounds for the level, but everything else would just fall into order.

(The reason you'd need a guard on the GetRegion call is that GetRegion is smart and, if you have a level with a pArr->x of 8, it counts -5, 3 and 11 as the same coordinate in order to allow wrap-around calculations elsewhere)

Reply:
i will try to make this change tomorrow